models/mask_rcnn.py

```diff
@@ -1259,11 +1261,21 @@ def load_from_pretrained(self):
       get_child(self, k).assign(v.numpy()).realize()
     return loaded_keys
 
-  def __call__(self, images):
+  def __call__(self, images, targets):
+    if self.training and targets is None:
+            raise ValueError("In training mode, targets should be passed")
     images = to_image_list(images)
     features = self.backbone(images.tensors)
-    proposals, _ = self.rpn(images, features)
-    x, result, _ = self.roi_heads(features, proposals)
+    proposals, proposal_losses = self.rpn(images, features, targets)
+    
+    x, result, detector_losses = self.roi_heads(features, proposals, targets)
+
+    if self.training:
+      losses = {}
+      losses.update(detector_losses)
+      losses.update(proposal_losses)
+      return losses
```

Comment:
I find this torch api to be a bit funky. I think it's a bit better to have the function always return a tuple, 

```
return result, losses if targets is not None else None
```

Then we would remove self.training altogether (pass targets=None if you don't want loss evaluation).

if no one disagrees I'll switch it.